atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -392,10 +391,11 @@ private enum State {
         this.transactionOutcomeMetrics =
                 TransactionOutcomeMetrics.create(transactionMetrics, metricsManager.getTaggedRegistry());
         this.expectationsDataCollectionMetrics = ExpectationsMetrics.of(metricsManager.getTaggedRegistry());
-        this.immutableTimestampLockManager = new ImmutableTimestampLockManager(
-                immutableTimestampLock, new DefaultLockValidityChecker(timelockService));
+        this.transactionLocksManager = new TransactionLocksManager(
+                immutableTimestampLock, new DefaultLockValidityChecker(timelockService), timelockService::tryUnlock);
+        closer.register(transactionLocksManager);
```

Comment:
See https://github.com/palantir/atlasdb/pull/7372/files, specifically https://github.com/palantir/atlasdb/pull/7372/files#diff-43819b9ee569afff73ea4707f9cfd11d683873ac3d695b54fd2f547c42f7da66L284.

Reply:
Ignoring the additional RPC if we do this. If this makes its way to timelock, and there was a previous successful unlock, it will just be ignored I think

https://github.com/palantir/atlasdb/blob/395fbf0f8ca751fed783e68b93ef75954864623b/timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/HeldLocksCollection.java#L105

Reply:
yeah double unlock should be safe (apart from perf)

Reply:
And this is going away on next PR